### ISSUE TITLE
JSII Kernel fills disk space using CDK on Cloud9

### ISSUE THREAD

**Issue by vaneek** [aws/jsii#700]
### Affected Languages

- [ ] `TypeScript` or `Javascript`
- [X] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)

### General Information
* **JSII Version:** 
```0.14.3```
* **Platform:**     
```Linux ip-172-31-27-45 4.14.133-88.105.amzn1.x86_64 #1 SMP Thu Jul 11 00:44:15 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux```

### What is the problem?

Every cdk execution creates a ```jsii-kernel-XXXXXX``` directory in ```/tmp``` eventually the filesystem fills up and ```/tmp``` must be manually cleaned.
### Verbose Log

https://gist.github.com/vaneek/1292391ce7a02fce283607c6d786b8ef

**Comment by RomainMuller** [comment]
Yeah I think we're not quite diligent enough when it comes to cleaning up temporary files behind ourselves. Thanks for reporting!

**Comment by jeshan** [comment]
It accumulates quite a lot of space (and not specific to Cloud9). I currently have 80 directories on my Linux Mint:

![image](https://user-images.githubusercontent.com/1020565/63803816-c73b4000-c926-11e9-8da9-0176613db7d6.png)

**Comment by jdiebold** [comment]
We experienced a similar problem when upgrading from 1.17.1 to 1.20.0. The jsii-kernel directories cluttered our tmp directy until our pod ran out of disk-space.
After the rollback to 1.17.1 it worked just fine.

**Comment by RomainMuller** [comment]
@jdiebold also using Python?

**Comment by Saad-Mohsin-Khan** [comment]
This error still persists I have a 1GB of EC2 instance and I get this error whenever I am running cdk synth and I also can't seem to find the jsii-kernel files in /tmp directory so that I can manually delete them myself.

**Comment by MartinLeedotOrg** [comment]
I'm seeing this in go as well:
github.com/aws/jsii-runtime-go v1.66.0

```
du -sch ./jsii-k*
262M    ./jsii-kernel-7uwjpU
58M     ./jsii-kernel-C5C15a
262M    ./jsii-kernel-I4aoXu
262M    ./jsii-kernel-IFBPyV
844M    total
```

**Comment by RomainMuller** [comment]
@MartinLeedotOrg in Go, do you start your `main` function with `defer jsii.Close()`? If not - does it help if you do?
